src/imitation/algorithms/adversarial/common.py

```diff
@@ -427,6 +428,44 @@ def _torchify_array(self, ndarray: Optional[np.ndarray]) -> Optional[th.Tensor]:
         if ndarray is not None:
             return th.as_tensor(ndarray, device=self.reward_train.device)
 
+    def _get_log_policy_act_prob(
+        self,
+        obs_th: th.Tensor,
+        acts_th: th.Tensor,
+    ) -> th.Tensor:
+        """Evaluates the given actions on the given observations.
+
+        Args:
+            obs_th: A batch of observations.
+            acts_th: A batch of actions.
+
+        Returns:
+            A batch of log policy action probabilities.
+        """
+        if isinstance(self.policy, policies.ActorCriticPolicy):
+            # policies.ActorCriticPolicy has a concrete implementation of
+            # evaluate_actions to generate log_policy_act_prob given obs and actions.
+            _, log_policy_act_prob_th, _ = self.policy.evaluate_actions(
+                obs_th,
+                acts_th,
+            )
+        elif isinstance(self.policy, sac_policies.SACPolicy):
+            gen_algo_actor = self.policy.actor
+            # generate log_policy_act_prob from SAC actor.
+            mean_actions, log_std, _ = gen_algo_actor.get_action_dist_params(obs_th)
+            distribution = gen_algo_actor.action_dist.proba_distribution(
+                mean_actions,
+                log_std,
+            )
+            # SAC applies a squashing function to bound the actions to a finite range
+            # `acts_th` need to be scaled accordingly before computing log prob.
+            scaled_acts_th = self.policy.scale_action(acts_th)
```

Comment:
This solution seems fine. It will break in environments that don't have bounded action spaces. But that's unusual, and there's not much reasonable we can do in that case.

Reply:
To be defensive, I might add an `assert self.policy.squash_output` as we'll be doing the wrong thing otherwise. It's currently hardcoded to True in SAC so we should be OK, but this feels like the kind of implementation detail that might change in the future.

Reply:
That's a good suggestion!